analytics/sensors/src/main/java/quarks/analytics/sensors/Deadtime.java

```diff
@@ -0,0 +1,98 @@
+package quarks.analytics.sensors;
+
+import java.util.Date;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import quarks.function.Predicate;
+
+/**
+ * A generic "deadtime" {@link Predicate}.
+ * <p>
+ * After accepting a tuple ({@link #test(Object) test()} returns true),
+ * any tuples received during the "deadtime" period are rejected
+ * ({@link #test(Object) test()} returns false).
+ * Then the next tuple is accepted and a new deadtime period begun.
+ * </p><p>
+ * The deadtime period may be changed while the topology is running
+ * via {@link #setPeriod(long, TimeUnit)}.
+ * </p>
+ *
+ * @param <T> tuple type
+ */
+public class Deadtime<T> implements Predicate<T> {
+    private static final long serialVersionUID = 1L;
+    private transient long deadtimePeriod;
+    private transient TimeUnit deadtimeUnit;
+    private transient long deadtimePeriodMillis;
```

Comment:
the same info is being stored twice, it seems to be only for the toString() method, is it worth it?